specs/Alert_Notifications.adoc

```diff
@@ -0,0 +1,189 @@
+
+
+= Alert Notifications
+
+Alerting system is essential to spot the problem and notify it to the corresponding user.
+The thresholds for alerts will be configured in grafana and the alerts will be raised by grafana.
+These alerts are captured by monitoring-integration module and notified with proper 
+details to respective user via the notifier.
+
+
+== Problem description
+
+The present alerting system monitors all the events and retrieves alert specific events.
+The alert specific events are processed and stored again in etcd under /alerting/alerts/
+as alerts recognized by tendrl. This leads to excess read and writes to etcd.
+Present implementation causes the alerting component to process events even
+when it can be done by a separate component before inserting the event in etcd.
+
+== Use Cases
+
+Goal: Notify user in case of any alert event.
+
+Actors:
+
+1) User
+
+Steps:
+
+* Node Agent stores alerting events in etcd.
+
+* These events will be fetched by the Notifier.
+
+* The alerting event is to be sent to respective user via configured
+  notification channels.
```

Comment:
Should be: `The alert notification is sent to the appropriate end points via corresponding notification channels.`

Alerts may not always end up directly with a user, they may be delivered to an intermediate system, eg. snmpd.